data-prepper-core/src/main/java/com/amazon/dataprepper/plugin/DefaultPluginFactory.java

```diff
@@ -0,0 +1,58 @@
+package com.amazon.dataprepper.plugin;
+
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.plugin.NoPluginFoundException;
+import com.amazon.dataprepper.model.plugin.PluginFactory;
+
+import java.util.Collection;
+import java.util.Objects;
+
+/**
+ * The primary implementation of {@link PluginFactory}.
+ *
+ * @since 1.2
+ */
+public class DefaultPluginFactory implements PluginFactory {
+
+    private final Collection<PluginProvider> pluginProviders;
+    private final PluginCreator pluginCreator;
+
+    public DefaultPluginFactory() {
+        this(new PluginProviderLoader(), new PluginCreator());
+    }
+
+    /**
+     * For testing only.
```

Comment:
These testing only constructors should be eliminated with the introduction of dependency injection. Let's add a TODO here.